src/common/HTMLFormatter.cc

```diff
@@ -88,6 +88,12 @@ void HTMLFormatter::output_header() {
   }
 }
 
+void HTMLFormatter::flush(std::ostream& os)
+{
+  m_pretty = false;
+  XMLFormatter::flush(os);
```

Comment:
this doesn't look right; we may call `flush()` several times during formatting of long documents, to avoid having to buffer the entire thing in memory. i don't think that should change the formatting